nevergrad/functions/images/imagelosses.py

```diff
@@ -1,35 +1,110 @@
+import cv2  # type: ignore
+import imquality.brisque as brisque  # type: ignore
+import lpips  # type: ignore
 import os
+import torch
 import typing as tp
 import numpy as np
 from nevergrad.functions.base import UnsupportedExperiment
+from nevergrad.common.decorators import Registry
+
+
+registry: Registry[tp.Any] = Registry()
 
 
 class ImageLoss:
     def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         pass
 
     def __call__(self, img: np.ndarray) -> float:
-        raise NotImplementedError
+        raise NotImplementedError(f"__call__ undefined in class {type(self)}")
 
 
-class SumAbsoluteDifferences(ImageLoss):
-    def __init__(self, reference: np.ndarray) -> None:
+class ImageLossWithReference(ImageLoss):
+    def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         if reference is None:
             raise ValueError("A reference is required")
         self.reference = reference
         super().__init__(reference)
+        assert len(self.reference.shape) == 3, self.reference.shape
+        assert self.reference.min() >= 0.0
+        assert self.reference.max() <= 255.0
+        assert self.reference.max() > 3.0  # Not totally sure but entirely black images are not very cool.
         self.domain_shape = self.reference.shape
 
+
+@registry.register
+class SumAbsoluteDifferences(ImageLossWithReference):
+    def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
+        super().__init__(reference)
+
     def __call__(self, x: np.ndarray) -> float:
         assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
         value = float(np.sum(np.fabs(x - self.reference)))
         return value
 
 
+@registry.register
+class Lpips(ImageLossWithReference):
+    def __init__(self, reference: np.ndarray, net: str) -> None:
```

Comment:
without default value, this breaks the API and probably the tests which goes through all the cases

Reply:
This class should not be registered, it makes no sense.
Your comment applies to other classes - I modify this.